src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -91,6 +92,44 @@ public static IEnumerable<string> GetResponseModelDefinitions(ServiceDefinition
             return respDefinitions;
         }
 
+        /// <summary>
+        /// Returns whether a string follows camel case style.
+        /// </summary>
+        /// <param name="name">String to check for style</param>
+        /// <returns>true if "name" follows camel case style, false otherwise.</returns>
+        public static bool isNameCamelCase(string name)
+        {
+            Regex propNameRegEx = new Regex(@"^[a-z0-9]+([A-Z][a-z0-9]+)+|^[a-z0-9]+$|^[a-z0-9]+[A-Z]$");
+            return (propNameRegEx.IsMatch(name));
+        }
+
+        /// <summary>
+        /// Returns a camel case styled string based on the string passed as parameter.
+        /// </summary>
+        /// <param name="name">String to convert to camel case style</param>
+        /// <returns>A string that conforms with camel case style based on the string passed as parameter.</returns>
+        public static string toCamelCase(string name)
+        {
+            StringBuilder sb = new StringBuilder(name);
+            if (sb.Length > 0)
+            {
+                sb[0] = sb[0].ToString().ToLower()[0];
+            }
+            bool firstUpper = true;
+            for (int i = 1; i < name.Length; i++)
+            {
+                if (char.IsUpper(sb[i]) && firstUpper)
+                {
+                    firstUpper = false;
+                }
+                else if (char.IsUpper(sb[i]))
```

Comment:
This worries me, proper camel casing should do "myBloatedPropertyName" i.e., subsequent capitalized letters should be okay http://stackoverflow.com/questions/5543490/json-naming-convention